sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp

```diff
@@ -0,0 +1,432 @@
+     * because the `m_endpoint` value is mocked on recorded clients.
+     */
+    void GetPolicyTest()
+    {
+      auto adminClient(CreateClient());
+
+      EXPECT_FALSE(adminClient->ClientVersion().empty());
+
+      AttestationType attestationType(GetParam().TeeType);
+      {
+        auto policy = adminClient->GetAttestationPolicy(attestationType);
+
+        // The policy should have a value, and the token should have been issued by the service.
+        // Note that if the policy *doesn't* have a body, then the attestation type must be TPM
+        // since TPM attestation is the only attestation type which allows empty policy
+        // documents.
+        if (policy.Value.Body.empty())
+        {
+          EXPECT_EQ(AttestationType::Tpm, attestationType);
+        }
+        else
+        {
+          EXPECT_EQ(0UL, policy.Value.Body.find("version"));
+        }
+
+        // In playback mode, the endpoint is a mocked value so the Issuer in the result will not
+        // match.
+        if (!m_testContext.IsPlaybackMode())
+        {
+          EXPECT_EQ(m_endpoint, policy.Value.Issuer.Value());
+        }
+      }
+
+      {
+        GetPolicyOptions gpOptions;
+        EXPECT_FALSE(gpOptions.TokenValidationOptions);
+      }
+    }
+
+    /** @brief Tests for policy modification using an unsecured JWS.
+     *
+     * Forwards to the `SetPolicyTest` and `ResetPolicyTest` with a non-present AttestationSigingKey
+     * parameter.
+     */
+    void ModifyPolicyUnsecuredTest()
+    {
+      SetPolicyTest();
+      ResetPolicyTest();
+    }
+
+    /** @brief Tests for policy modification using a secured JWS with an emphemerally generated key.
+     *
+     * Forwards to the `SetPolicyTest` and `ResetPolicyTest` with a newly created
+     * AttestationSigingKey parameter.
+     */
+    void ModifyPolicySecuredTest()
+    {
+      auto rsaKey(Cryptography::CreateRsaKey(2048));
+      auto signingCert(
+          Cryptography::CreateX509CertificateForPrivateKey(rsaKey, "CN=TestSetPolicyCertificate"));
+
+      auto signingKey(
+          AttestationSigningKey{rsaKey->ExportPrivateKey(), signingCert->ExportAsPEM()});
+
+      SetPolicyTest(signingKey);
+      ResetPolicyTest(signingKey);
+    }
+
+    /** @brief Tests for policy modification using a secured JWS with a predefined key.
+     *
+     * Forwards to the `SetPolicyTest` and `ResetPolicyTest` with an AttestationSigingKey parameter
+     * defined in new-testresources.ps1.
+     *
+     * Note that this is a live-only test, because
+     */
+    void ModifyPolicyIsolatedTest()
+    {
+      // In PlaybackMode, the values of ISOLATED_SIGNING_CERTIFICATE and ISOLATED_SIGNING_KEY are
+      // replaced with dummy values which cannot be converted into actual certificates. So skip
+      // the isolated mode tests when we are in playback or record mode (there's no point in
+      // recording something that cannot work).
+      if (m_testContext.IsLiveMode())
```

Comment:
:(

I would remove this check and mark all test who would call this method as `_LIVE_ONLY_` . 

But that's a nit.  I remember all the things you mentioned about Attestation and test cases...  So, this is just me trying to have a separation between test-fw and service's tests.  Where the service's tests consumes API from test-fw and avoid consuming its state ( or even learn about the test-fw concepts )

Reply:
If I remove the check, all my tests become live_only, and that means that I never test my client in recorded tests. 

And that seems to be a bigger issue than disabling two or three checks on live tests.